storage2/src/storage.rs

```diff
@@ -59,15 +57,16 @@ impl Storage {
     /// `Storage`, or `None` if the tree is empty.
     pub async fn latest_version(&self) -> Result<Option<jmt::Version>> {
         // TODO: do better
-        Ok(latest_version(self.0.db).unwrap())
+        latest_version(&self.0.db)
     }
 
     /// Returns a new [`State`] on top of the latest version of the tree.
-    pub async fn state(&self) -> State {
+    pub fn state(&self) -> State {
```

Comment:
This can be non-`async`, since we're just cloning the snapshot handle we already have.